### ISSUE TITLE
If SpaceMarineMissileBarrage is charged when SpaceMarine dies, server crashes next turn

### ISSUE THREAD

**Issue by Zheneq** [Zheneq/hc#19]

**Comment by Zheneq** [comment]
```
2023-01-15 19:21:59.024+00:00 [WRN] Trying to get free pos wrt a null square (IsDead: True).  for Totter#750 Code issue-- actor probably instantiated but not on Board yet. (warningid 2fbe6ea4)
2023-01-15 19:21:59.025+00:00 [WRN] NullReferenceException: Object reference not set to an instance of an object (warningid 44316430)
   at SpaceMarineMissileBarrageEffect.GatherEffectResults (.EffectResults& effectResults, Boolean isReal)
   at Effect.GatherEffectResults_Base (AbilityPriority phase, Boolean isReal)
   at Effect.Resolve ()
   at GameFlow.GatherActionsInPhase (.ServerActionBuffer actionBuffer, AbilityPriority phase, System.Collections.Generic.List`1& executingPlayerActions)
   at GameFlow.HandleUpdateResolve ()
   at GameFlow.Update_FCFS ()
   at GameFlow.Update ()
2023-01-15 19:21:59.025+00:00 [WRN] OnUpdatePhaseEndedNoServerPlayer (warningid 31a6480c)
2023-01-15 19:21:59.039+00:00 [INF] Going to next turn ability phase DEPRICATED_Combat_Charge
2023-01-15 19:21:59.039+00:00 [INF] New phase DEPRICATED_Combat_Charge of turn 17, 3 requests
2023-01-15 19:21:59.076+00:00 [WRN] ArgumentOutOfRangeException: Argument is out of range. (warningid dad84df7)
Parameter name: index
   at System.Collections.Generic.List`1[Theatrics.Phase].get_Item (Int32 index)
   at Theatrics.Turn.AssignAnimationPlayOrderAndCinematicsForPhase (AbilityPriority phasePriority, Boolean mergeBounds, Boolean showDebugInfo)
   at Theatrics.Turn.SetupAbilityPhase (AbilityPriority phasePriority, System.Collections.Generic.List`1 abilityRequests, System.Collections.Generic.HashSet`1 hitActorIds, Boolean hasHitsWithoutAnimEntry)
   at TheatricsManager.SetupTurnAbilityPhase (AbilityPriority phasePriority, System.Collections.Generic.List`1 abilityRequests, System.Collections.Generic.HashSet`1 hitActorIds, Boolean hasHitsWithoutAnimEntry)
   at GameFlow.HandleUpdateResolve ()
   at GameFlow.Update_FCFS ()
   at GameFlow.Update ()
```

**Comment by Zheneq** [comment]
Fixed in a40c16ee9aa8305b473c40e1afb28c87c0615ea2
